communication/nats/dialog/dialog_waiter_test.go

```diff
@@ -186,9 +186,15 @@ type mockedIdentityRegistry struct {
 	anyIdentityRegistered bool
 }
 
+// IsRegistered mock
 func (mir *mockedIdentityRegistry) IsRegistered(address common.Address) (bool, error) {
 	return mir.anyIdentityRegistered, nil
 }
 
+// WaitForRegistrationEvent mock
+func (mir *mockedIdentityRegistry) WaitForRegistrationEvent(providerAddress common.Address, registeredEvent chan int, stopLoop chan int) {
```

Comment:
This function signature is a bit complicated, some parameters are IN type (stopLoop which is modified from outside), others are OUT (registeredEvent channel which is modified inside function)
I suggest the following signature -> SubscribeToRegistrationEvent(identityAddress) returns registeredEvent chan of type (RegisteredEvent not int), Unsubscribe func(), error (in case of any error). That way caller can always cance subscription if needed and check for any errors. Also - unsubscribe function hides any internal details how subscribtion is really implemented